Whitespace removed around standalone macro with attribute

When a using standalone macro with an attribute on it, swift-format wants to remove whitespace around the macro.

The following example assumes the default swift-format configuration. Tested on 509.0.0 and `main`.

**Input:**

```swift
@available(iOS 17.0, *)
#Preview("Name") {
  EmptyView()
}
```

**Expected output:** same as input

**Actual output:**

```
Test.swift:1:24: warning: [RemoveLine] remove line break
Test.swift:2:17: warning: [Spacing] remove 1 space
```

```swift
@available(iOS 17.0, *)#Preview("Name"){
  EmptyView()
}
```

--- a/Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift
+++ b/Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift
@@ -1324,6 +1324,12 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
   }
 
   override func visit(_ node: MacroExpansionDeclSyntax) -> SyntaxVisitorContinueKind {
+    arrangeAttributeList(node.attributes)
+
+    before(
+      node.trailingClosure?.leftBrace,
+      tokens: .break(.same, newlines: .elective(ignoresDiscretionary: true)))
+
     arrangeFunctionCallArgumentList(
       node.arguments,
       leftDelimiter: node.leftParen,

--- a/Tests/SwiftFormatTests/PrettyPrint/MacroCallTests.swift
+++ b/Tests/SwiftFormatTests/PrettyPrint/MacroCallTests.swift
@@ -114,4 +114,16 @@ final class MacroCallTests: PrettyPrintTestCase {
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 45)
   }
+
+  func testMacroDeclWithAttributesAndArguments() {
+    let input = """
+      @nonsenseAttribute
+      @available(iOS 17.0, *)
+      #Preview("Name") {
+        EmptyView()
+      }
+      
+      """
+    assertPrettyPrintEqual(input: input, expected: input, linelength: 45)
+  }
 }

